packages/agent/src/oidc-v2.ts

```diff
@@ -0,0 +1,689 @@
+  presentation_definition?: any;
+
+  /** A JSON object containing the Verifier metadata values (Tim's note: from TBD KCC Repo) */
+  client_metadata?: {
+    /** Array of strings, each a DID method supported for the subject of ID Token	*/
+    subject_syntax_types_supported: string[];
+    /** Human-readable string name of the client to be presented to the end-user during authorization */
+    client_name?: string;
+    /** URI of a web page providing information about the client */
+    client_uri?: string;
+    /** URI of an image logo for the client */
+    logo_uri?: string;
+    /** Array of strings representing ways to contact people responsible for this client, typically email addresses */
+    contacts?: string[];
+    /** URI that points to a terms of service document for the client */
+    tos_uri?: string;
+    /** URI that points to a privacy policy document */
+    policy_uri?: string;
+  };
+};
+
+/**
+ * Claims specific to Web5 Connect rather than OIDC
+ * The contents of this are inserted into a JWT inside of the {@link PushedAuthRequest}.
+ */
+export type Web5ConnectRequest = {
+  /** PermissionGrants that are to be sent to the provider */
+  permission_requests: ConnectPermissionRequests;
+};
+
+// old, ignore, do not use. here for comparison.
+// export interface AuthorizationRequestObject {
+//   claims?: { id_token: { delegation_grants: any; [key: string]: any } };
+//   client_id: string;
+//   client_metadata?: { client_uri?: string; [key: string]: any };
+//   code_verifier?: string;
+//   nonce: string;
+//   redirect_uri: string;
+//   state?: string;
+//   [key: string]: any;
+// }
+
+/** An auth response that is compatible with both Web5 Connect and (hopefully, WIP) OIDC SIOPv2 */
+export type HybridAuthResponse = SIOPv2AuthResponse & Web5ConnectAuthResponse;
+
+/** The fields for an OIDC SIOPv2 Auth Repsonse */
+export type SIOPv2AuthResponse = {
+  /** Issuer Identifier for the Issuer of the response. */
+  iss: 'https://self-issued.me/v2';
+  /** Subject Identifier. A locally unique and never reassigned identifier
+   * within the Issuer for the End-User, which is intended to be consumed
+   * by the Client. */
+  sub: string;
+  /** Audience(s) that this ID Token is intended for. It MUST contain the
+   * OAuth 2.0 client_id of the Relying Party as an audience value. */
+  aud: string;
+  /** Time at which the JWT was issued. */
+  iat: number;
+  /** Expiration time on or after which the ID Token MUST NOT be accepted
+   * for processing. */
+  exp: number;
+  /** Time when the End-User authentication occurred. */
+  auth_time?: number;
+  /** b64url encoded nonce used to associate a Client session with an ID Token, and to
+   * mitigate replay attacks. */
+  nonce?: string;
+  /** Custom claims. */
+  [key: string]: any;
+};
+
+/** The fields for an Web5 Connect Auth Response */
+export type Web5ConnectAuthResponse = {
+  delegation_grants: any[];
```

Comment:
`delegatedGrants` to align with casing style of `privateKeyJwks`.

Reply:
done